HACKING.md

```diff
@@ -0,0 +1,88 @@
+# Hacking
```

Comment:
I don't have strong feelings on this, but from my experience, it seems like people are moving towards `CONTRIBUTING.md` rather than `HACKING.md` for this sort of content. It also seems like this is more about contributing than just generally hacking around with the code.

On the other hand, `HACKING.md` is nice synchronicity with ops...

Reply:
+1 on `CONTRIBUTING.md`

Reply:
Since we've got a plus one on CONTRIBUTING, CONTRIBUTING it is then :)